src/ResourceManagement/Insights/Microsoft.Azure.Insights/Generated/Management/Insights/Models/MetricTrigger.cs

```diff
@@ -42,13 +42,13 @@ public partial class MetricTrigger
         /// that is collected should be combined over time. The default value
         /// is Average. Possible values include: 'Average', 'Minimum',
         /// 'Maximum', 'Total', 'Count'</param>
-        /// <param name="operatorProperty">the operator that is used to
-        /// compare the metric data and the threshold. Possible values
-        /// include: 'Equals', 'NotEquals', 'GreaterThan',
-        /// 'GreaterThanOrEqual', 'LessThan', 'LessThanOrEqual'</param>
+        /// <param name="operatorProperty">the operator that is used to compare
+        /// the metric data and the threshold. Possible values include:
+        /// 'Equals', 'NotEquals', 'GreaterThan', 'GreaterThanOrEqual',
+        /// 'LessThan', 'LessThanOrEqual'</param>
         /// <param name="threshold">the threshold of the metric that triggers
         /// the scale action.</param>
-        public MetricTrigger(string metricName = default(string), string metricResourceUri = default(string), System.TimeSpan? timeGrain = default(System.TimeSpan?), MetricStatisticType? statistic = default(MetricStatisticType?), System.TimeSpan? timeWindow = default(System.TimeSpan?), TimeAggregationType? timeAggregation = default(TimeAggregationType?), ComparisonOperationType? operatorProperty = default(ComparisonOperationType?), double? threshold = default(double?))
+        public MetricTrigger(string metricName, string metricResourceUri, System.TimeSpan timeGrain, MetricStatisticType statistic, System.TimeSpan timeWindow, TimeAggregationType timeAggregation, ComparisonOperationType operatorProperty, double threshold)
```

Comment:
This is also correcting a bug in the Spec. All these fields are required in an autoscale metric trigger.
They are already declared as mandatory in PowerShell's New-AzureRmAutoscaleRule cmdlet.